src/charm.py

```diff
@@ -82,15 +80,12 @@ def _configure(self):
         # check if configuration file has changed and if so push the
         # new config file to the workload container
         prometheus_config = self._prometheus_config()
-        config_hash = hashlib.md5(str(prometheus_config).encode("utf-8")).hexdigest()
-        if self._stored.prometheus_config_hash != config_hash:
-            try:
-                container.push(PROMETHEUS_CONFIG, prometheus_config)
-                self._stored.prometheus_config_hash = config_hash
-                logger.info("Pushed new configuration")
-            except ConnectionError:
-                logger.info("Ignoring config changed since pebble is not ready")
-                return
+        try:
+            container.push(PROMETHEUS_CONFIG, prometheus_config)
```

Comment:
comment is outdated: no check anymore
but I don't understand:
1. why push in the constructor
2. how will you know to restart/hot-reload?

Reply:
Hot reloading is moot point anyway. Prometheus requires a `HUP`, and Pebble does not (yet) provide a way to run arbitrary commands inside the sidecar

Reply:
If you expand a bit, you can see this is not in the constructor